src/vm/object.cpp

```diff
@@ -1776,9 +1775,10 @@ VOID Object::ValidateInner(BOOL bDeep, BOOL bVerifyNextHeader, BOOL bVerifySyncB
         // try to validate next object's header
         if (bDeep 
             && bVerifyNextHeader 
-            && GCScan::GetGcRuntimeStructuresValid ()
+            && GCHeapUtilities::IsGCHeapInitialized()
```

Comment:
IsGCHeapInitialized call does not look necessary - if we have an object to validate, it sure must have come from GCHeap and thus the GCHeap is initialized.